localization/util/pose_initializer/src/pose_initializer_core.cpp

```diff
@@ -167,11 +172,10 @@ bool PoseInitializer::getHeight(
   if (map_ptr_) {
     tf2::Transform transform;
     try {
-      const auto stamped =
-        tf2_buffer_.lookupTransform(map_frame_, fixed_frame, ros::Time(0), ros::Duration(1.0));
+      const auto stamped = tf2_buffer_.lookupTransform(map_frame_, fixed_frame, tf2::TimePointZero);
```

Comment:
There is a slight semantic difference here – the timeout doesn't matters if the tf buffer contains at least one pose, no matter the timestamp. If this timeout is here intentionally for the case where the tf buffer might not contain any pose, please let me know.

Reply:
I think this is okay if we are using Time(0) since it is looking for the newest transform.
However, I think we have some nodes that require lookupTransform with exact time, which will need some timeout.
If you have any experience in using tf2::BufferClient (which seems to have lookupTransform with timeout), could you comment on this issue?https://github.com/tier4/Pilot.Auto/issues/23

Reply:
I think even with Time(0) there is a corner case where a timeout is relevant: If an initial pose message or service call occurs slightly before whatever node is responsible has published the first transform, it will fail, i.e. throw an exception.